mqtt/mqtt-broker/src/auth/authentication.rs

```diff
@@ -13,38 +12,56 @@ pub enum Credentials {
 }
 
 /// Represents a client certificate.
-#[derive(Clone, Debug, From)]
+#[derive(Clone, Debug)]
 pub struct Certificate(Vec<u8>);
 
+impl From<Vec<u8>> for Certificate {
+    fn from(certificate: Vec<u8>) -> Self {
+        Self(certificate)
+    }
+}
+
 /// A trait to authenticate a MQTT client with given credentials.
 #[async_trait]
 pub trait Authenticator {
+    /// Authentication error.
+    type Error;
```

Comment:
should we force this to be `std::error::Error`? The broker below has a type bound on `Display` which is so close to being `std::error::Error` (just need to add `source`)

Reply:
```rust
type Error: std::error::Error;
```